src/io.rs

```diff
@@ -72,11 +74,50 @@ impl OutputBuilder {
                 err_fn,
             ),
         }
-        .unwrap();
+    }
+}
 
-        stream.play().unwrap();
+impl Default for OutputBuilder {
+    fn default() -> Self {
+        let host = cpal::default_host();
+        let device = host
+            .default_output_device()
+            .expect("no output device available");
 
-        stream
+        let mut supported_configs_range = device
+            .supported_output_configs()
+            .expect("error while querying configs");
+
+        let supported_config = if cfg!(target_os = "linux") {
+            device
+                .default_output_config()
+                .expect("default config not found")
+        } else {
+            supported_configs_range
+                .next()
+                .expect("no supported config?!")
+                .with_max_sample_rate()
+        };
+
+        let sample_format = supported_config.sample_format();
+
+        // determine best buffer size. Spec requires BUFFER_SIZE, but that might not be available
+        let mut buffer_size = match supported_config.buffer_size() {
+            SupportedBufferSize::Range { min, .. } => crate::BUFFER_SIZE.max(*min),
```

Comment:
I'm worried some backends don't support any multiple of 128 and this will fail too.
Maybe in the `default` implementation, we simply do not fiddle with the buffer_size and let cpal choose it. Combined with your other work in #7 that should work 100% of the time

Reply:
A large cpal default buffer size will cost a delayed start but after that the rendering should be as responsive as the rendering buffer size namely 128 (if i understand it correctly).

I think you're right we should go directly on default config for every host. 

If cpal makes Fixed behavior more consistent between host in the future, this should be reconsidered.

Reply:
I change my mind, i let fixed buffer size. still possible to change if bugs are reported. For now, it works and give the best perfs.\
let me know what you think